Makefile.build

```diff
@@ -577,6 +576,11 @@ kernel/kernel.cma: kernel/kernel.mllib
 	$(SHOW)'OCAMLC -a -o $@'
 	$(HIDE)$(OCAMLC) $(MLINCLUDES) $(BYTEFLAGS) $(VMBYTEFLAGS) -a -o $@ $(filter-out %.mllib, $^)
 
+# Specific rule for kernel.cmxa as to adjoin -cclib -lcoqrun
+kernel/kernel.cmxa: kernel/kernel.mllib
+	$(SHOW)'OCAMLOPT -a -o $@'
+	$(HIDE)$(OCAMLOPT) $(MLINCLUDES) $(OPTFLAGS) -I kernel/byterun/ -cclib -lcoqrun -a -o $@ $(filter-out %.mllib, $^)
```

Comment:
It's not new in this PR, but why do we use `filter-out` instead of `$<`?

Reply:
No idea.

Reply:
I'm a bit rusty, why only the .cmxa archive needs the extra flag? and not the .cma ?

Reply:
The `cma` file gets the `VMBYTEFLAGS`; the way C code is linked in bytecode builds is a bit different c.f. #6475 , so I dunno what the right magic here should be.

Reply:
So likely the `-cclib` flag only works in bytecode if using `-custom` as in native.